pkg/cmd/admin/migrate/images/images.go

```diff
@@ -0,0 +1,484 @@
+	var changed bool
+	for k, v := range cfg {
+		original := k
+		if updateString(&k, fn) {
+			changed = true
+			delete(cfg, original)
+			cfg[k] = v
+		}
+	}
+	return changed
+}
+
+// ImageReferenceMapping represents a transformation of an image reference.
+type ImageReferenceMapping struct {
+	FromRegistry string
+	FromName     string
+	ToRegistry   string
+	ToName       string
+}
+
+// ParseMapping converts a string in the form "(REGISTRY|*)/(NAME|*)" to an ImageReferenceMapping
+// or returns a user-facing error. REGISTRY is the image registry value (hostname) or "docker.io".
+// NAME is the full repository name (the path relative to the registry root).
+// TODO: handle v2 repository names, which can have multiple segments (must fix
+//   ParseDockerImageReference)
+func ParseMapping(s string) (ImageReferenceMapping, error) {
+	parts := strings.SplitN(s, "=", 2)
+	from := strings.SplitN(parts[0], "/", 2)
+	to := strings.SplitN(parts[1], "/", 2)
+	if len(from) < 2 || len(to) < 2 {
+		return ImageReferenceMapping{}, fmt.Errorf("all arguments must be of the form REGISTRY/NAME=REGISTRY/NAME, where registry or name may be '*' or a value")
+	}
+	if len(from[0]) == 0 {
+		return ImageReferenceMapping{}, fmt.Errorf("%q is not a valid source: registry must be specified (may be '*')", parts[0])
+	}
+	if len(from[1]) == 0 {
+		return ImageReferenceMapping{}, fmt.Errorf("%q is not a valid source: name must be specified (may be '*')", parts[0])
+	}
+	if len(to[0]) == 0 {
+		return ImageReferenceMapping{}, fmt.Errorf("%q is not a valid target: registry must be specified (may be '*')", parts[1])
+	}
+	if len(to[1]) == 0 {
+		return ImageReferenceMapping{}, fmt.Errorf("%q is not a valid target: name must be specified (may be '*')", parts[1])
+	}
+	if from[0] == "*" {
+		from[0] = ""
+	}
+	if from[1] == "*" {
+		from[1] = ""
+	}
+	if to[0] == "*" {
+		to[0] = ""
+	}
+	if to[1] == "*" {
+		to[1] = ""
+	}
+	if to[0] == "" && to[1] == "" {
+		return ImageReferenceMapping{}, fmt.Errorf("%q is not a valid target: at least one change must be specified", parts[1])
+	}
+	if from[0] == from[1] && to[0] == to[1] {
+		return ImageReferenceMapping{}, fmt.Errorf("%q is not valid: must target at least one field to change", s)
+	}
+	return ImageReferenceMapping{
+		FromRegistry: from[0],
+		FromName:     from[1],
+		ToRegistry:   to[0],
+		ToName:       to[1],
+	}, nil
+}
+
+// ImageReferenceMappings provide a convenience method for transforming an input reference
+type ImageReferenceMappings []ImageReferenceMapping
+
+// MapReference transforms the provided Docker image reference if any mapping matches the
+// input. If the reference cannot be parsed, it will not be modified.
+func (m ImageReferenceMappings) MapReference(in string) string {
+	ref, err := imageapi.ParseDockerImageReference(in)
+	if err != nil {
+		return in
+	}
+	registry := ref.Registry
+	if len(registry) == 0 {
+		registry = "docker.io"
+	}
```

Comment:
In this case I'm avoiding changing other fields, and I don't want the other DockerClientDefaults (which include tags).  I'll use the constant though.